src/components/NavBar/NavigationBar.tsx

```diff
@@ -0,0 +1,167 @@
+"use client";
+
+import React from "react";
+import AppBar from "@mui/material/AppBar";
+import IconButton from "@mui/material/IconButton";
+import MenuIcon from "@mui/icons-material/Menu";
+import SwipeableDrawer from "@mui/material/SwipeableDrawer";
+import Link from "next/link";
+import styled from "styled-components";
+import Button from "@mui/material/Button";
+import LightDarkSlider from "@/components/NavBar/LightDarkSlider";
+import SignOutButton from "@/components/NavBar/SignOutButton";
+import LinkButton from "@/components/Buttons/LinkButton";
+
+export const PageButton = styled(Button)`
+    color: white;
+    font-size: 1.25rem;
+    margin: 10px;
+    &:hover {
+        background-color: ${(props) => props.theme.primaryBackgroundColor};
+    }
+`;
+
+const UnstyledLink = styled(Link)`
+    text-decoration: none;
+    display: contents;
+`;
+
+const Logo = styled.img`
+    max-width: 100%;
+    max-height: 100%;
+    align-content: center;
+`;
+
+const AppBarInner = styled.div`
+    display: flex;
+    height: 4rem;
+    padding: 0.5rem;
+`;
+
+const DrawerInner = styled.div`
+    display: flex;
+    flex-direction: column;
+    justify-content: start;
+    align-items: stretch;
+    padding: 1rem;
+    width: 15rem;
+`;
+
+const Gap = styled.div`
+    width: 1rem;
+`;
+
+const DrawerButtonWrapper = styled.div`
+    padding: 1rem;
+    border-bottom: 1px solid ${(props) => props.theme.secondaryBackgroundColor};
+
+    &:last-child {
+        border-bottom: none;
+    }
+`;
+
+const DrawerButton = styled(Button)`
+    color: ${(props) => props.theme.secondaryBackgroundColor};
+    width: 100%;
+`;
+
+const StickyAppBar = styled(AppBar)`
+    position: sticky;
+`;
+
+const NavElementContainer = styled.div`
+    display: flex;
+    flex-basis: 0;
```

Comment:
I cant figure out why this flex basis is here. Can you explain why?

Reply:
So it helps with the flexboxing. Essentially it sets all of the 4 (3 visible at a time) container divs to 0px to start with, so that when `flex-grow : 1` is then applied they grow to the same size.